### PR TITLE
🥳🥳🥳 android work 🥳🥳🥳

### PR DESCRIPTION
**Pull request by osdnk**
All tasks on linear

- [x] Optimize / rethink transaction list
- [x] Optimize wallet list
- [x] Build Bottom sheet #1086
- [x] Big delays on press #1156
- [x] Fix touchables 85762a1
- [x] Optimize Modals: #1131 + 81de398 + #1134 + #1156
- [x] Fix status bar
- [x] Paddings / margins
- [x] Rethink emojis
- [x] Rethink navigation patterns
- [x] Splash screen
- [x] Savings 
- [x] Crashes - maybe #1130
- [x] Keyboard
- [x] Backup sheet
- [x] Startup time
- [x] Welcome Screen Animated Button
- [x] Send Sheet
- [x] Google login backdrop
- [x] Back up manually 
- [x] Review
- [x] custom gas

### CODE REVIEW

File: android/app/src/main/java/me/rainbow/MainActivity.java [hunk 1]
```diff
@@ -6,11 +6,13 @@
...
 import org.devio.rn.splashscreen.SplashScreen;
 import io.branch.rnbranch.*;
+import me.rainbow.NativeModules.RNBackHandler.RNBackHandlerPackage;
+
 import android.content.Intent;
 
 public class MainActivity extends ReactActivity {
 
-  @Override
+  @Override
```
Comment (jinchung) [inline android/app/src/main/java/me/rainbow/MainActivity.java:15]: trailing space

File: scripts/postinstall.sh [hunk 2]
```diff
@@ -25,7 +25,7 @@ then
 
 else
   echo "⚠️ .env file missing!! ⚠️"
-  echo "Please make sure the file exists and it's located in the root of the project"
+  echo "Please make sure the file exists and it's located in the root of  the project"
```
Comment (jinchung) [inline scripts/postinstall.sh:28]: this change not needed

File: .eslintrc.json [hunk 3]
```diff
@@ -13,7 +13,10 @@
   },
   "globals": {
     "__DEV__": true,
-    "IS_DEV": true
+    "IS_DEV": true,
+    "ios": true,
+    "android": true,
+    "akd": true
```
Comment (jinchung) [inline .eslintrc.json:19]: any reason we would prefer not to name this `androidKeyboardDismiss` for clarity?

Comment (osdnk) [reply .eslintrc.json:19]: hm, i made it like that and didn't revisit. will fix

File: scripts/postinstall.sh (hunk 2 above)
Comment (osdnk) [reply scripts/postinstall.sh:28]:
```suggestion
  echo "Please make sure the file exists and it's located in the root of the project"
```

File: android/app/src/main/java/me/rainbow/MainActivity.java (hunk 1 above)
Comment (osdnk) [reply android/app/src/main/java/me/rainbow/MainActivity.java:15]:
```suggestion
  @Override
```

File: src/react-native-yet-another-bottom-sheet/Example/bottom-sheet.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import React, { useEffect } from 'react';
```
Comment (jinchung) [inline src/react-native-yet-another-bottom-sheet/Example/bottom-sheet.js:1]: Do we need this Example file?

Comment (osdnk) [reply src/react-native-yet-another-bottom-sheet/Example/bottom-sheet.js:1]: no

File: src/handlers/cloudBackup.js [hunk 5]
```diff
@@ -22,8 +21,13 @@ export const CLOUD_BACKUP_ERRORS = {
   WALLET_BACKUP_STATUS_UPDATE_FAILED: 'Update wallet backup status failed',
 };
 
+export function logoutFromGoogleDrive() {
+  android && RNCloudFs.logout();
+}
+
 // This is used for dev purposes only!
 export async function deleteAllBackups() {
+  await RNCloudFs.loginIfNeeded();
```
Comment (jinchung) [inline src/handlers/cloudBackup.js:30]: all the other uses of loginIfNeeded first check if android, not sure if we care to have it here too

File: src/hooks/useAccountTransactions.js [hunk 6]
```diff
@@ -17,9 +19,17 @@ export default function useAccountTransactions(initialized, isFocused) {
     })
   );
 
+  const [page, setPage] = useState(1);
+  const nextPage = useCallback(() => setPage(page => page + 1), [setPage]);
```
Comment (jinchung) [inline src/hooks/useAccountTransactions.js:23]: Don't need setPage in dependencies list

File: src/screens/ImportSeedPhraseSheet.js [hunk 7]
```diff
@@ -233,9 +232,15 @@ export default function ImportSeedPhraseSheet() {
       try {
         setBusy(true);
         setTimeout(async () => {
-          const { hdnode, isHDWallet, type, wallet } = getWallet(input);
-          setCheckedWallet({ hdnode, isHDWallet, type, wallet });
-          const ens = await web3Provider.lookupAddress(wallet?.address);
+          const type = identifyWalletType(input);
+          let walletResult;
+          if (type === WalletTypes.privateKey) {
```
Comment (jinchung) [inline src/screens/ImportSeedPhraseSheet.js:237]: Can we just use the ethereumUtils.js `deriveAccountFromMnemonicOrPrivateKey` function here in this case?

File: src/hooks/useWalletsDebug.js [hunk 8]
```diff
@@ -84,15 +85,26 @@ export default function useWalletsDebug() {
           await dispatch(walletsSetSelected(currentWallet));
           await dispatch(walletsLoadState());
 
-          let wallet;
+          let walletResult;
           if (type === WalletTypes.privateKey) {
-            wallet = new Wallet(addHexPrefix(seedObject.seedphrase));
+            walletResult = ethereumUtils.deriveAccountFromPrivateKey(
```
Comment (jinchung) [inline src/hooks/useWalletsDebug.js:90]: Could also use ethereumUtils `deriveAccountFromMnemonicOrPrivateKey` here as well

File: src/utils/ethereumUtils.js [hunk 9]
```diff
@@ -210,8 +225,51 @@ const checkIfUrlIsAScam = async url => {
...
+  return {
+    address: ethersWallet.address,
+    isHDWallet: false,
+    root: null,
+    type: WalletTypes.privateKey,
+    wallet: ethersWallet,
+    walletType: WalletLibraryType.ethers,
+  };
+};
+const deriveAccountFromMnemonicOrPrivateKey = mnemonicOrPrivateKey => {
```
Comment (jinchung) [inline src/utils/ethereumUtils.js:261]: can we just have a line break here between the functions

**Review by jinchung — APPROVED** [review]
Only had minor comments - approved! 🌮

**Comment by osdnk** [comment]
@brunobar79 pls halp with conflicts

**Review by skylarbarrera — APPROVED** [review]
LGTM! 🥳

**Pull request merged**